BUG: unable to build with musl, 'execinfo.h' file not found

### Describe the issue:

Unable to build with musl since 'execinfo.h' not exists.

### Reproduce the code example:

```python
N/A
```


### Error message:

```shell
INFO: clang: numpy/core/src/multiarray/temp_elide.c
numpy/core/src/multiarray/temp_elide.c:85:10: fatal error: 'execinfo.h' file not found
#include <execinfo.h>
                    ^~~~~~~~~~~~
1 error generated.
```
```


### NumPy/Python version information:

1.23

--- a/numpy/core/setup_common.py
+++ b/numpy/core/setup_common.py
@@ -142,7 +142,9 @@ OPTIONAL_HEADERS = [
                 "immintrin.h",  # AVX
                 "features.h",  # for glibc version linux
                 "xlocale.h",  # see GH#8367
-                "dlfcn.h", # dladdr
+                "dlfcn.h",  # dladdr
+                "execinfo.h",  # backtrace
+                "libunwind.h",  # backtrace for LLVM/Clang using libunwind
                 "sys/mman.h", #madvise
 ]
 

--- a/numpy/core/src/multiarray/temp_elide.c
+++ b/numpy/core/src/multiarray/temp_elide.c
@@ -82,7 +82,12 @@
 #define NPY_MIN_ELIDE_BYTES (32)
 #endif
 #include <dlfcn.h>
+
+#if defined HAVE_EXECINFO_H
 #include <execinfo.h>
+#elif defined HAVE_LIBUNWIND_H
+#include <libunwind.h>
+#endif
 
 /*
  * linear search pointer in table

--- a/numpy/linalg/umath_linalg.cpp
+++ b/numpy/linalg/umath_linalg.cpp
@@ -50,7 +50,11 @@ using dispatch_scalar = typename std::conditional<std::is_scalar<typ>::value, sc
     } while (0)
 
 #if 0
+#if defined HAVE_EXECINFO_H
 #include <execinfo.h>
+#elif defined HAVE_LIBUNWIND_H
+#include <libunwind.h>
+#endif
 void
 dbg_stack_trace()
 {

